src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -3690,15 +3711,28 @@ scanlMAfter' fstep initial done s =
 
 {-# INLINE scanl' #-}
 scanl' :: Monad m => (b -> a -> b) -> b -> Stream m a -> Stream m b
-scanl' f = scanlM' (\a b -> return (f a b))
+scanl' f seed = scanlM' (\a b -> return (f a b)) (return seed)
 
 {-# INLINE_NORMAL scanlM #-}
-scanlM :: Monad m => (b -> a -> m b) -> b -> Stream m a -> Stream m b
-scanlM fstep begin s = begin `cons` postscanlM fstep begin s
+scanlM :: Monad m => (b -> a -> m b) -> m b -> Stream m a -> Stream m b
+scanlM fstep begin (Stream step state) = Stream step' Nothing
```

Comment:
Why not use `begin 'consM' postscanlM fstep begin s`?

Reply:
No reason in particular

Reply:
I guess it cannot be written like that, how will you get the initial state `s` of `postscanlM`? You want to execute the action only once.